### ISSUE TITLE
PDF annotation popup tooltips do not show up

### ISSUE THREAD

**Issue by tobidelbruck** [sumatrapdfreader/sumatrapdf#2083]
SumatraPDF no longer shows PDF annotation popup tooltips that is expected to pop up as they do in Adobe Reader or Acrobat.  SumatraPDF version 3.0.0 showed them fine.

SumatraPDF v3.3.3 does not show the tip which Acrobat shows when hovering over "BRAM" as seen in screen shot below

![2021-08-01_102134](https://user-images.githubusercontent.com/10061874/127764488-01389d8a-b213-4f5a-a6c2-e30f76e32499.png)

See https://www.overleaf.com/read/fyvbnxpwwjss for overleaf project that generates a test PDF with embedded popup tooltip.

The resulting PDF is at https://www.dropbox.com/s/sc2jacft6huva3d/acronyms_with_tooltip_test.pdf?dl=0

**Comment by GitHubRulesOK** [comment]
@kjk 
Note these files are built in LaTeX via several methods thus there can be variable results, in this case it starts
\include{acronyms} 
which preloads
\usepackage[draft]{pdfcomment} 

Another more common related package is "HyperRef" which builds internal Hyperlinks that are more frequently encountered and a separate issue.

Above quoted compiled sample is
[acronyms_with_tooltip_test.pdf](https://github.com/sumatrapdfreader/sumatrapdf/files/6912686/acronyms_with_tooltip_test.pdf)

**Comment by kjk** [comment]
Yeah, for tooltip_test.pdf, mupdf complains that "cannot create appearance stream for Screen annotations".

**Comment by tobidelbruck** [comment]
Thanks. Please note that the acronyms.tex "package" that I hacked together only puts popup tooltips on 2nd or later use of the acronym. Not the first one. In acrobat DC 2021.005.20060 all tips in acronyms_with_tooltip_test.pdf show properly.

**Comment by tobidelbruck** [comment]
Still open in 3.4. Please let me know how/if I can help. What is MuPDF? How can I help revert the regression from 3.0 to later versions? 3.0 still works fine for these popup annotations on acronyms, as does latest Acrobat DC Reader and editor.

**Comment by kjk** [comment]
https://mupdf.com/ is the code sumatra uses for PDF handling.

They have a simple windows viewer (https://mupdf.com/releases/index.html) you can use to double-check if a given issue is specific to Sumatra or is an issue with mupdf itself.

If it happens in mupdf, it helps if you report the bug to them https://bugs.ghostscript.com/

If they fix it, we'll inherit the fix.

**Comment by 5inf** [comment]
The LaTeX pdfcomment package [1] has a good example document [2] which can be used for testing. None of the tooltips in the example document work at the moment in SumatraPDF 3.3.3. They however work fine in 3.0.

Also the tooltips created by the cooltooltips package [3], which seem to be different again because they work in e.g. in the included PDF viewer of TeXstudio, are not working in 3.3.3 and 3.0.

mupdf-1.19.0 behaves exactly as SumatraPDF 3.3.3.

I have however not found a version of mupdf that behaves like SumatraPDF3.0.

[1] https://www.ctan.org/pkg/pdfcomment
[2] https://ftp.rrzn.uni-hannover.de/pub/mirror/tex-archive/macros/latex/contrib/pdfcomment/doc/pdfcomment_de.pdf
[3] https://www.ctan.org/pkg/cooltooltips

**Comment by tobidelbruck** [comment]
> The LaTeX pdfcomment package [1] has a good example document [2] which can be used for testing. None of the tooltips in the example document work at the moment in SumatraPDF 3.3.3. They however work fine in 3.0.
> 
> Also the tooltips created by the cooltooltips package [3], which seem to be different again because they work in e.g. in the included PDF viewer of TeXstudio, are not working in 3.3.3 and 3.0.
> 
> mupdf-1.19.0 behaves exactly as SumatraPDF 3.3.3.
> 
> I have however not found a version of mupdf that behaves like SumatraPDF3.0.
> 
> [1] https://www.ctan.org/pkg/pdfcomment [2] https://ftp.rrzn.uni-hannover.de/pub/mirror/tex-archive/macros/latex/contrib/pdfcomment/doc/pdfcomment_de.pdf [3] https://www.ctan.org/pkg/cooltooltips

Same for me; I tried downloading all versions of muPDF from 1.19.0 back to 1.9a from https://mupdf.com/releases/index.html and none of them show any popup tooltip with hover or rollover of the annotation. So I don't know how SumatraPDF3.0.0 could show the tips.  The latest version of muPDF when built in WSL2 and displayed with GWSL x-server on windows 10 not show any hover/rollover tooltip popups in any of these versions.
However I do see some big code edits in https://github.com/ArtifexSoftware/mupdf/blob/master/source/pdf/pdf-form.c (the acronyms are highlighted as forms in muPDF)

**Comment by tobidelbruck** [comment]
Note, could be related, overleaf had a problem with white boxes showing for these tooltip acronyms until they released a fix yesterday. Now the popups work properly again in the integrated overleaf PDF viewer.

**Comment by tobidelbruck** [comment]
The overleaf fix is not related: I got this email response from overleaf team; it might provide some clue but it appears they use the Mozilla PDF rendering engine:

> On Overleaf, this was related to changes/bugs in pdf.js: https://github.com/mozilla/pdf.js/pull/14247
> 
> We're not that familiar with muPDF, but if it uses its own engine and unrelated to pdf.js, then the issue in SumatraPDF is likely due to a different cause. I'm afraid we're unable to delve further into SumatraPDF; sorry about this!
>

**Comment by GitHubRulesOK** [comment]
Similar to https://github.com/sumatrapdfreader/sumatrapdf/issues/2391
![image](https://user-images.githubusercontent.com/12754913/229222283-443a6873-1cec-40d7-b811-f5aea84a8531.png)

**Comment by tobidelbruck** [comment]
See https://github.com/tobidelbruck/acronyms-with-tooltips-latex for dual-annotation workaround for missing muPDF feature

**Comment by GitHubRulesOK** [comment]
@tobidelbruck 
it is best not to relate behaviour differences to other PDF.JS issues as the one you linked in 2022 was linked back to https://bugzilla.mozilla.org/show_bug.cgi?id=1737260 which was about OCG behaviour which is "layer controls" and that example CGI bug is shown here as a difference in MuPDF controls 
<img width="818" height="426" alt="Image" src="https://github.com/user-attachments/assets/629fd4fb-c6c0-422d-ba87-e481e26e0a85" />

Your OP test file in Acrobat powered Edge shows a slight mismatch but works there within reason in MuPDF it would be significantly different as it uses the annotation layer.
<img width="364" height="106" alt="Image" src="https://github.com/user-attachments/assets/d406db02-16a7-4f19-90d6-6cdfb090d810" />
<img width="679" height="242" alt="Image" src="https://github.com/user-attachments/assets/3c41286d-9d47-4eb5-aa7b-c929e78bd85a" />

However with current SumatraPDF the tooltip caries the comment and better than mupdf if links are on shows the area correctly thus in my view the original issue is closed ?

<img width="800" height="400" alt="Image" src="https://github.com/user-attachments/assets/b33f0c22-e7bf-4ef9-ab61-e05683beceb6" />

<img width="339" height="126" alt="Image" src="https://github.com/user-attachments/assets/084962d4-1dad-47ab-b55d-ff1874eda5ac" />

**Comment by GitHubRulesOK** [comment]
@kjk the OP issue was addressed some time ago but clouded by other different commentary thus they should be separate issues.

<img width="637" height="183" alt="Image" src="https://github.com/user-attachments/assets/512a569d-dfeb-48c5-a6bd-5a4fd993f114" />

**Comment by tobidelbruck** [comment]
See https://github.com/tobidelbruck/acronyms-with-tooltips-latex and output overleaf project https://www.overleaf.com/project/610656ba700ab464a2e45b06

This hack (to overlay 2 types of annotations on the same acronym) makes tooltips show up on more PDF viewers. I believe Sumatra 3.7.21481 (latest as of today) still uses the squarebox overlay to show the annotation. See comment in https://github.com/tobidelbruck/acronyms-with-tooltips-latex/blob/main/acronymtooltips-pdfcomment.tex:

% Dual tooltips: original pdfcomment /Btn /TU widget + /Square /Contents markup.
% Viewers differ — Acrobat favors /Btn; SumatraPDF/muPDF favor /Square. Both overlap.
%
% Square popup: omit /Subj; combined height, depth 0 (cleaner popup in some viewers).

Anyhow, this type of dual tooltip works on Sumatra 3.7. But Sumatra still does not show the Adobe "standard" /Btn /TU widget popup, as far as I know.
